sound/soc/sof/amd/acp.c

```diff
@@ -484,10 +509,28 @@ static int acp_init(struct snd_sof_dev *sdev)
 	return acp_reset(sdev);
 }
 
+static bool check_acp_sdw_enable_status(struct snd_sof_dev *sdev)
+{
+	struct acp_dev_data *acp_data;
+	u32 sdw0_en, sdw1_en;
+
+	acp_data = sdev->pdata->hw_pdata;
+	if (!acp_data->sdw)
+		return false;
+
+	sdw0_en = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_SW0_EN);
+	sdw1_en = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_SW1_EN);
+	acp_data->sdw_en_stat = sdw0_en || sdw1_en;
+	return acp_data->sdw_en_stat;
+}
+
 int amd_sof_acp_suspend(struct snd_sof_dev *sdev, u32 target_state)
 {
 	int ret;
 
+	if (check_acp_sdw_enable_status(sdev))
+		return acp_dsp_reset(sdev);
```

Comment:
you may want to elaborate with a comment on the relationship between DSP and SoundWire. IIRC the first generation did not use the DSP so that's a bit confusing.

You have a paragraph in the commit message that I think is related

"When SoundWire configuration is enabled, In case of ClockStopMode,
DSP soft reset should be applied and for rest of the scenarios
acp init/deinit sequence should be invoked."

but this test doesn't exactly make a clear reference to clock stop mode.

Reply:
will add a comment in the code